### PR TITLE
BUG: Add ``is_any_real_numeric_dtype`` to solve discrepancy between ```Index.is_numeric()``` and ```is_numeric_dtype()```

### LINKED ISSUE #51152: BUG: Discrepancy between `.is_numeric()` and `is_numeric_dtype` for `bool` `Index`
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd
from pandas.api.types import is_numeric_dtype

idx = pd.Index([True], dtype=bool)

print(f"{idx.is_numeric() = }")
print(f"{is_numeric_dtype(idx) = }")
```

### Issue Description

With the latest nightly wheel of `pandas`, the following deprecation message is given

```python
FutureWarning: Index.is_numeric is deprecated. Use pandas.api.types.is_numeric_dtype instead
```

However, when accounting for this deprecation over in `dask/dask`, we observed that `is_numeric()` and `is_numeric_dtype` don't give the same result for and `Index` with `bool` data

### Expected Behavior

I'd expect these to both give the same answer 

### Installed Versions

<details>

```
INSTALLED VERSIONS
------------------
commit           : f06c96a93fb2e21c9f801192d9ed5896c5ce3535
python           : 3.10.4.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 22.2.0
Version          : Darwin Kernel Version 22.2.0: Fri Nov 11 02:08:47 PST 2022; root:xnu-8792.61.2~4/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 2.0.0.dev0+1401.gf06c96a93f
numpy            : 1.24.1
pytz             : 2022.1
dateutil         : 2.8.2
setuptools       : 59.8.0
pip              : 22.0.4
Cython           : None
pytest           : 7.1.3
hypothesis       : None
sphinx           : 4.5.0
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : 1.1
pymysql          : None
psycopg2         : None
jinja2           : 3.1.2
IPython          : 8.2.0
pandas_datareader: None
bs4              : 4.11.1
bottleneck       : None
brotli           :
fastparquet      : 2022.12.1.dev6
fsspec           : 2023.1.0+5.g012816b
gcsfs            : None
matplotlib       : 3.5.1
numba            : None
numexpr          : 2.8.0
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 11.0.0.dev316
pyreadstat       : None
pyxlsb           : None
s3fs             : 2022.10.0
scipy            : 1.9.0
snappy           :
sqlalchemy       : 1.4.35
tables           : 3.7.0
tabulate         : None
xarray           : 2022.3.0
xlrd             : None
zstandard        : None
tzdata           : None
qtpy             : None
pyqt5            : None
```

</details>

### PR DESCRIPTION
**Pull request by ABCPAN-rank**
- [x] closes #51152 
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/v2.0.0.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by phofl** [comment]
Could you check the failing builds?

File: doc/source/whatsnew/v2.0.0.rst [hunk 1]
```diff
@@ -761,6 +761,7 @@ Other API changes
 - The levels of the index of the :class:`Series` returned from ``Series.sparse.from_coo`` now always have dtype ``int32``. Previously they had dtype ``int64`` (:issue:`50926`)
 - :func:`to_datetime` with ``unit`` of either "Y" or "M" will now raise if a sequence contains a non-round ``float`` value, matching the ``Timestamp`` behavior (:issue:`50301`)
 - The methods :meth:`Series.round`, :meth:`DataFrame.__invert__`, :meth:`Series.__invert__`, :meth:`DataFrame.swapaxes`, :meth:`DataFrame.first`, :meth:`DataFrame.last`, :meth:`Series.first`, :meth:`Series.last` and :meth:`DataFrame.align` will now always return new objects (:issue:`51032`)
+- Added :func:`pandas.api.types.is_any_real_numeric_dtype` to instead of :meth:`Index.is_numeric`(:issue:`51152`)
```
Comment (phofl) [inline doc/source/whatsnew/v2.0.0.rst:764]: can you reword to:

Added :func:`pandas.api.types.is_any_real_numeric_dtype` to check for real numeric dtypes (:issue:`51152`)

File: pandas/core/dtypes/api.py [hunk 2]
```diff
@@ -68,6 +69,7 @@
     "is_named_tuple",
     "is_number",
     "is_numeric_dtype",
+    "is_any_real_numeric_dtype",
```
Comment (phofl) [inline pandas/core/dtypes/api.py:72]: Can you keep this in order? Alphabetically speaking

File: doc/source/reference/arrays.rst [hunk 3]
```diff
@@ -666,6 +666,7 @@ Data type introspection
     api.types.is_integer_dtype
     api.types.is_interval_dtype
     api.types.is_numeric_dtype
+    api.types.is_any_real_numeric_dtype
```
Comment (phofl) [inline doc/source/reference/arrays.rst:669]: Same as below, would prefer to keep alphabetical

File: ci/code_checks.sh [hunk 4]
```diff
@@ -608,6 +608,7 @@ if [[ -z "$CHECK" || "$CHECK" == "docstrings" ]]; then
         pandas.api.types.is_list_like \
         pandas.api.types.is_named_tuple \
         pandas.api.types.is_numeric_dtype \
+        pandas.api.types.is_any_real_numeric_dtype \
```
Comment (phofl) [inline ci/code_checks.sh:611]: Same here

File: doc/source/whatsnew/v2.0.0.rst [hunk 5]
```diff
@@ -775,7 +776,7 @@ Deprecations
 - :meth:`Index.is_integer` has been deprecated. Use :func:`pandas.api.types.is_integer_dtype` instead (:issue:`50042`)
 - :meth:`Index.is_floating` has been deprecated. Use :func:`pandas.api.types.is_float_dtype` instead (:issue:`50042`)
 - :meth:`Index.holds_integer` has been deprecated. Use :func:`pandas.api.types.infer_dtype` instead (:issue:`50243`)
-- :meth:`Index.is_numeric` has been deprecated. Use :func:`pandas.api.types.is_numeric_dtype` instead (:issue:`50042`)
+- :meth:`Index.is_numeric` has been deprecated. Use :func:`pandas.api.types.is_any_real_numeric_dtype` instead (:issue:`51152`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.0.0.rst:779]:
```suggestion
- :meth:`Index.is_numeric` has been deprecated. Use :func:`pandas.api.types.is_any_real_numeric_dtype` instead (:issue:`50042`, :issue:`51152`)
```

Good to keep the old reference too

File: pandas/core/dtypes/common.py [hunk 6]
```diff
@@ -1219,9 +1219,9 @@ def is_numeric_dtype(arr_or_dtype) -> bool:
     )
 
 
-def is_any_numeric_dtype(arr_or_dtype) -> bool:
+def is_any_real_numeric_dtype(arr_or_dtype) -> bool:
```
Comment (mroeschke) [inline pandas/core/dtypes/common.py:1222]: @phofl on the off chance someone is actually using this, maybe it's safer if we make an alias `is_any_real_numeric_dtype = is_any_numeric_dtype` the "public" version?

Comment (phofl) [reply pandas/core/dtypes/common.py:1222]: It was only added a couple of days ago, it's especially in no release yet, so I think we are good.

Comment (mroeschke) [reply pandas/core/dtypes/common.py:1222]: Ah okay gotcha. Fine to change then

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @ABCPAN-rank

**Pull request merged**